denovogear/denovogear
# Feature unittest find mutation

#### Add unittest for FindMutation class
- This is part of the multi PR mentioned in issue #161. The next PR will move some stats from `FindMutation` class to `MutationStats` class.
- This PR add some unittest for the FindMutation class. Some tests depend on the results from mutation.cc, and the unittest for mutation.cc is not completed yet.
- Add fixture `TrioWorkspace`, which automatically setup tri workspace for the test.
- Update `BOOST_ABS_THRESHOLD`. Some of these expected values are 4-6 d.p. only, and these are used in `BamTest.cmake.in`. These summary stats are tested with `BOOST_CHECK_SMALL` instead of `_CLOSE`. This is relate to #157.
- Several minor changes/clean up to make this PR work properly.
